codspeed 3.0.0 doesn't compile on windows

https://github.com/facet-rs/facet/actions/runs/15972291441/job/45046025039?pr=825

Highlight:

```
error[E0432]: unresolved import `nix::libc`
 --> C:\Users\Administrator\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\codspeed-3.0.0\src\fifo.rs:3:10
  |
3 | use nix::libc::O_NONBLOCK;
  |          ^^^^ could not find `libc` in `nix`

error[E0432]: unresolved import `nix::sys`
 --> C:\Users\Administrator\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\codspeed-3.0.0\src\fifo.rs:4:10
  |
4 | use nix::sys::stat;
  |          ^^^ could not find `sys` in `nix`

```

and:

```
error[E0599]: no method named `custom_flags` found for mutable reference `&mut OpenOptions` in the current scope
   --> C:\Users\Administrator\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\codspeed-3.0.0\src\fifo.rs:113:18
    |
110 | /             OpenOptions::new()
111 | |                 .write(true)
112 | |                 .read(true)
113 | |                 .custom_flags(O_NONBLOCK)
    | |                 -^^^^^^^^^^^^ method not found in `&mut OpenOptions`
    | |_________________|
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -19,6 +19,19 @@ jobs:
         with:
           extra_args: --all-files
 
+  lint-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: moonrepo/setup-rust@v1
+        with:
+          components: rustfmt, clippy
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --all-files
+
   msrv-check:
     runs-on: ubuntu-latest
     steps:
@@ -109,6 +122,7 @@ jobs:
     if: always()
     needs:
       - lint
+      - lint-windows
       - msrv-check
       - tests
       - compat-integration-test-instrumentation

